apps/users-service/src/routes/authRoutes.ts

```diff
@@ -0,0 +1,107 @@
+import { Request, Response } from 'express';
+const express = require('express');
+const bcrypt = require('bcryptjs');
+const { query } = require('../database/connection');
+
+const router = express.Router();
+
+// POST /api/auth/signup
+router.post('/signup', async (req: Request, res: Response) => {
```

Comment:
Let's add a zod middleware here to validate the input, which will throw an error in case something is missing, or does not fit the schema:
```suggestion
// zod middleware
const validate = ({ schema }) => (req, res) => {
  // validate the schema, throw error if it does not adhere
}

router.post(
  '/signup', 
  validate({ schema: {
     body: z.object({ 
        email: z.email(),
        password: z.string().min(10), // or whatever password constraints we want
        firstName: z.string(),
        lastName: z.string(),
        }),
  }), async (req: Request, res: Response) => {
```